include/ringmesh/mesh/cartesian_grid.h

```diff
@@ -281,11 +281,126 @@ namespace RINGMesh
+            }
+            else
+            {
+                throw RINGMeshException(
+                    "CartesianGrid", "Give a valid attribute type." );
+            }
+        }
+
+        /*!
+         * Creates and adds a new attribute to the cartesian grid from its
+         * name and values on the grid.
+         * \param[in] attribute_name the name of the new attribute to add
+         * \param[in] values a vector of the values of the attribute on all
+         * the cartesian grid cells.
+         */
+        template < typename T >
+        void add_attribute(
+            const std::string& attribute_name, const std::vector< T >& values )
+        {
+            if( values.size() != nb_total_cells_ )
+            {
+                throw RINGMeshException( "CartesianGrid",
+                    "The attribute you're trying to add doesn't have the same"
+                    " number of values than the grid." );
+            }
+            attributes_manager_.bind_attribute_store( attribute_name,
+                GEO::AttributeStore::
+                    create_attribute_store_by_element_type_name(
+                        type_to_string< T >(), 1 ) );
+            GEO::Attribute< T > attribute{ attributes_manager_,
+                attribute_name };
+            for( auto i : range( nb_total_cells_ ) )
+            {
+                attribute[i] = values[i];
+            }
+        }
+
+        /*!
+         * Creates a new attribute with the name \attribute_name and
+         * the same value \value on every grid point.
+         */
+        template < typename T >
+        void add_attribute( const std::string& attribute_name, T value = 0 )
+        {
+            attributes_manager_.bind_attribute_store( attribute_name,
+                GEO::AttributeStore::
+                    create_attribute_store_by_element_type_name(
+                        type_to_string< T >(), 1 ) );
+            GEO::Attribute< T > attribute{ attributes_manager_,
+                attribute_name };
+            attribute.fill( value );
+        }
+
+        /*!
+         * Returns a pointer to an Attribute of the grid, from its name
+         * \attribute_name.
+         */
+        template < typename T >
+        GEO::Attribute< T >* get_attribute( const std::string& attribute_name )
+        {
+            return new GEO::Attribute< T >(
+                attributes_manager_, attribute_name );
+        }
+
+        /*!
+         * Returns the value of the attribute \attribute_name at the grid
+         * position \position.
+         */
+        template < typename T >
+        T& get_attribute_value(
+            const std::string& attribute_name, sivecn< DIMENSION > position )
+        {
+            index_t index = cell_offset( position );
+            if( index == NO_ID )
+            {
+                throw RINGMeshException( "CartesianGrid",
+                    "Points outside of the grid have no attribute value, give "
```

Comment:
Clarify the output message using grid dimension/range (think about "out of range").